python/ray/serialization.py

```diff
@@ -158,18 +158,50 @@ def actor_handle_deserializer(serialized_obj):
 
             def id_serializer(obj):
                 if isinstance(obj, ray.ObjectID) and obj.is_direct_call_type():
-                    obj = self.worker.core_worker.promote_object_to_plasma(obj)
+                    obj = self.worker.core_worker.serialize_object_id(obj)
                 return pickle.dumps(obj)
 
             def id_deserializer(serialized_obj):
                 return pickle.loads(serialized_obj)
 
+            def object_id_serializer(obj):
```

Comment:
Please change the naming of serialize/deserialize to make clear they have side-effects, e.g., deserialize_and_register.